illumos-utils/src/opte/non_illumos/port_manager.rs

```diff
@@ -137,47 +191,50 @@ impl PortManager {
                 "IPv6 is not yet supported for guest interfaces",
             ))),
         }?;
-        let boundary_services = default_boundary_services();
-        let port_name = self.inner.next_port_name();
-        let vnic = format!("v{}", port_name);
-        let (port, ticket) = {
-            let mut ports = self.inner.ports.lock().unwrap();
-            let ticket = PortTicket::new(
-                instance_id,
-                port_name.clone(),
-                self.inner.clone(),
-            );
-            let port = Port::new(
-                port_name.clone(),
-                nic.ip,
-                subnet,
-                mac,
-                nic.slot,
-                vni,
-                self.inner.underlay_ip,
-                source_nat,
-                external_ips,
-                gateway,
-                boundary_services,
-                vnic,
-            );
-            let old =
-                ports.insert((instance_id, port_name.clone()), port.clone());
-            assert!(
-                old.is_none(),
-                "Duplicate OPTE port detected: instance_id = {}, port_name = {}",
-                instance_id,
-                &port_name,
-            );
-            (port, ticket)
-        };
 
-        info!(
-            self.inner.log,
-            "Created OPTE port for guest";
-            "port" => ?&port,
-        );
-        Ok((port, ticket))
+        self.create_port_inner(
+            instance_id,
+            PortType::Guest { slot: nic.slot },
+            None,
+            ip,
+            mac,
+            vni,
+            gateway,
+            firewall_rules,
+            external_ips.is_some(),
+        )
+    }
+
+    pub fn create_svc_port(
+        &self,
+        svc_id: Uuid,
+        svc_name: &str,
+        mac: MacAddr6,
+        _external_ip: IpAddr,
+        vni: u32,
+        firewall_rules: &[VpcFirewallRule],
+    ) -> Result<(Port, PortTicket), Error> {
+        // TODO: Assume each service zone has a single port.
+        let port_name = format!("{}_{}0", XDE_LINK_PREFIX, svc_name);
+
+        // TODO: Arbitrary subnet, should come from a "service" VPC.
+        let vpc_subnet: Ipv4Network = "10.0.0.0/24".parse().unwrap();
```

Comment:
same question here - this could be wrapped up in a NetworkInterface

Reply:
done